Sources/Algorithms/Stride.swift

```diff
@@ -0,0 +1,112 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// striding(by:)
+//===----------------------------------------------------------------------===//
+
+extension Collection {
+  /// - Complexity: O(_1_) and access to the next value in the striding
+  /// is O(_1_) if the collection conforms to `RandomAccessCollection`,
+  /// otherwise O(_k_) where `k` is the striding `step`.
```

Comment:
```suggestion
  /// - Complexity: O(1). Access to successive values is O(1) if the
  /// collection conforms to `RandomAccessCollection`; otherwise,
  /// O(_k_), where _k_ is the striding `step`.
```

Also, the documentation should start with a single sentence describing what the method returns.